docs/SphinxReplayDB.md

```diff
@@ -0,0 +1,165 @@
+   - The immediate previous hop (David)
+   - Cannot determine the original sender
+
+4. **Onion Encryption**: Each hop peels one layer of encryption, revealing only
+   the next hop's information
+
+
+The detailed mechanics are described in [BOLT 04](https://github.com/lightning/bolts/blob/master/04-onion-routing.md)
+
+
+## Replaying Onion Packets
+
+The [sphinx protocol](https://cypherpunks.ca/~iang/pubs/Sphinx_Oakland09.pdf)
+clearly states that implementations of the protocol should guard against 
+replay attacks.
+Replaying (resending) onion packets into the network can compromise the
+privacy guarantees promised by the protocol. So it is crucial for node
+participants to not forward replayed onion packets to guard the privacy of all
+network participants. Compared to the original Sphinx protocol, the Lightning
+Network has an improved replay protection in place especially when it comes to
+forwarding HTLCs, which are different from Onion Messages introduced later on
+because HTLCs lock a payment to the particular onion packet. Therefore, sending
+HTLCs packets comes with a cost. Moreover, every HTLC has an expiry date, also
+called CLTV (absolute locktime), which prevents the replay of packets that have
+already expired. In addition to the CLTV expiry of a packet, every HTLC onion
+packet commits to the payment hash in the HMAC of the message (to be precise,
+the associated data), so this prevents an attacker from attaching an old onion
+packet to a new payment hash, which now also comes with the risk for the
+attacker that he does not only have to lock funds when replaying an onion
+packet but he also risks that the next node settles the HTLC because it already
+knows the preimage of the HTLC. Although the attack comes with a high cost,
+Lightning implementations should prevent replayed onion packets from
+propagating through the network to safeguard the privacy for every network
+participant.
+
+## Replay attack
+
+An attacker could execute a re-injection attack by strategically positioning
+well-connected nodes within the network, ensuring they participate in a high
+volume of payment forwarding. In this scenario, the attacker's nodes act as
+forwarding intermediaries. These malicious nodes would collect passing onion
+packets and concurrently monitor network gossip. Upon observing a successful
+payment forward, the attacker could re-inject the captured onion message. By
+then monitoring channel updates or closures from the neighboring node, the
+attacker aims to determine if that channel was part of the payment path.
+Furthermore, if the neighboring node, lacking a suitable outgoing channel,
+instead settles the payment (i.e., acts as the final recipient), it becomes
+highly probable that it was the ultimate receiving node. It's important to note,
+however, that such an attack carries significant costs.
+
+### Replay Protection in LND
+
+For that reason the LND implements the replay protection of those replayed onion
```

Comment:
We can remove this line all together from this section..."For that reason the LND implements the replay protection of those replayed onion packets."

Reply:
done!